uint8 stateMask
bool ping
